src/components/Thought.js

```diff
@@ -435,7 +446,20 @@ const ThoughtContainer = ({
       dragPreview(getEmptyImage())
     }
   }}>
-    <div className='thought-container'>
+    <div className='thought-container'
+      onMouseDown={e => {
+        // Used timeout to prevent turning lightblue on click
+        const timeoutId = setTimeout(() => {
+          store.dispatch({ type: 'dragInProgress', value: true, draggedThoughtsRanked: thoughtsRankedLive })
+          alert('Drag and drop to move thought')
+        }, TIMEOUT_BEFORE_DRAG)
+        store.dispatch({ type: 'dragTimeoutId', timeoutId })
```

Comment:
I think we can store `dragTimeoutId` locally rather than add an additional global reducer. I'm not sure whether closure, `useState` or `useEffect` is the right solution. 

@shresthabijay What do you think?

Reply:
@raineorshine Yes it's better to store `dragTimeoutId` locally. I recommend creating a custom hook that manages long press event.

```javascript
// useLongPress.js custom hook to manage long press
import { useState, useEffect, useCallback, useRef } from 'react'

export default function useLongPress(onLongPressStart = () => {}, onLongPressEnd = () => {}, ms = 250) {
  const [startLongPress, setStartLongPress] = useState(false)
  const timerIdRef = useRef()

  useEffect(() => {
    if (startLongPress) timerIdRef.current = setTimeout(onLongPressStart, ms)
    else clearTimeout(timerIdRef.current)

    return () => clearTimeout(timerIdRef.current)
  }, [startLongPress])

  const start = useCallback(() => {
    setStartLongPress(true)
  }, [])

  const stop = useCallback(() => {
    setStartLongPress(false)
    onLongPressEnd()
  }, [])

  return {
    onMouseDown: start,
    onMouseUp: stop,
    onTouchStart: start,
    onTouchEnd: stop,
  }
}

```

Now we use this custom hook wherever we need long press event.

```javascript

// inside ThoughtContainer component 
 const onLongPressStart = () => {
    store.dispatch({ type: 'dragInProgress', value: true, draggedThoughtsRanked: thoughtsRankedLive })
    alert('Drag and drop to move thought')
  }

  const onLongPressEnd = () => {
    store.dispatch({ type: 'dragInProgress', value: false })
    alert(null)
  }

  const longPressHandlerProps = useLongPress(onLongPressStart, onLongPressEnd, TIMEOUT_BEFORE_DRAG)

  // pass long press handler props to the target div
  return (
    <div className='thought-container' {...longPressHandlerProps}></div>
   )
```
This way we can abstract long press logic properly in separate file without needing to handle timer id inside `ThoughtContainer`.

Reply:
Yeah, this looks great